### DISCUSSION TITLE
Slow Internet connection inside containers

### DISCUSSION THREAD

**Discussion by 0-wHiTeHand-0** [nextcloud/all-in-one#4682] [Questions] [answered]
Hi,
I have built Nextcloud AIO in my VPS (Debian, 4 cores & 16Gb RAM) and it works pretty well. Communications between containers work and from the host to Internet also work. However, when making a request from a container to Internet, for example from nextcloud-aio-nextcloud:

```
$ time curl https://nextcloud.com > /dev/null
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  163k    0  163k    0     0  11788      0 --:--:--  0:00:14 --:--:-- 11059
real	0m 14.24s
user	0m 0.02s
sys	0m 0.02s
```

It takes 14.24 seconds. But from the host:

```
# time curl https://nextcloud.com > /dev/null
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  163k    0  163k    0     0  98561      0 --:--:--  0:00:01 --:--:-- 98541

real	0m1.714s
user	0m0.037s
sys	0m0.004s
```

From the host takes 1.714 seconds.

This problem is causing timeouts when Nextcloud tries to download "big" (not too big) files like https://apps.nextcloud.com/api/v1/apps.json (~14Mb). I've had to download apps.json in the host, and set a temporary python HTTP server to serve the file internally using the "appstoreurl" parameter in config.php. I use this as a temporary workaround, but obviously it's not a good solution.

However, if i upload a big file to Nextcloud or if I use Talk with video (things that cause high network activity), it works perfectly well. I use an NGINX reverse proxy, so the 'external' communication is done through it.

I've tried to change the internal Docker DNSs ( https://forums.docker.com/t/internet-access-in-docker-container-is-much-slower-compared-to-access-from-machine-running-docker/141076 and https://dockerlabs.collabnix.com/intermediate/networking/Configuring_DNS.html ) but it's not a DNS problem.
I've checked the MTUs, but none of them is under 1500.
I've reviewed https://github.com/nextcloud/all-in-one/discussions/1335 , I've reviewed https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#6-how-to-debug-things , I don't use Cloudflare, I've checked that the brute-force protection is not throttling requests, and I don't know what else can I check.

Any idea about what can be wrong in the Docker configuration?

Thanks!

Update:
It seems that packets are being lost at some point between the host and the container. Ping from the container:
```
$ ping apps.nextcloud.com
PING apps.nextcloud.com (176.9.217.53): 56 data bytes
64 bytes from 176.9.217.53: seq=0 ttl=42 time=306.184 ms
64 bytes from 176.9.217.53: seq=1 ttl=42 time=306.115 ms
...
^C
--- apps.nextcloud.com ping statistics ---
22 packets transmitted, 19 packets received, 13% packet loss
round-trip min/avg/max = 297.202/305.103/307.062 ms
```
There's no packet loss if the Ping is sent from the host.

**Comment by szaimen** [comment, marked as answer]
Hi, I fear I cannot really help any further. Looks like you already did quite some investigation on the topic. Possibly it makes sense to ask in a docker forum for further helps as it looks like a docker networking issue...

**Comment by bf-whiteh** [comment]
Yes, in my opinion bothering other people should be the last option; asking things that are already responded publicly is a waste of time for everybody. So, I've tried everything I found before asking.
Anyways, I will go to the Docker forum; if I manage to fix this I will post the solution here, so it can be helpful to other people.
Thanks!
